src/ui/EditorNS/editor.cpp

```diff
@@ -23,53 +24,61 @@ namespace EditorNS
     Editor::Editor(QWidget *parent) :
         QWidget(parent)
     {
-
-        QString themeName = NqqSettings::getInstance().Appearance.getColorScheme();
-        if (themeName == "") {
-            themeName = "default";
-        }
-
-        fullConstructor(themeFromName(themeName));
+        fullConstructor();
     }
 
-    Editor::Editor(const Theme &theme, QWidget *parent) :
-        QWidget(parent)
+    void Editor::fullConstructor()
     {
-        fullConstructor(theme);
-    }
+        auto& s = NqqSettings::getInstance();
+        /* Initialize some values here so we don't have issues*/
+        const auto language = "default";
+        setIndentationMode(!s.Languages.getIndentWithSpaces(language),
+                            s.Languages.getTabSize(language),
+                            false,
+                            false);
 
-    void Editor::fullConstructor(const Theme &theme)
-    {
         initJsProxy();
-        initWebView(theme);
+        initWebView();
         initContextMenu();
-        m_layout = new QVBoxLayout(this);
-        m_layout->setContentsMargins(0, 0, 0, 0);
-        m_layout->setSpacing(0);
-        m_layout->addWidget(m_webView, 1);
-        setLayout(m_layout);
-        
+
+        auto vbox = new QVBoxLayout(this);
+        vbox->setContentsMargins(0, 0, 0, 0);
+        vbox->setSpacing(0);
+        vbox->addWidget(m_webView, 1);
+        setLayout(vbox);
+
         connect(m_webView->page(),
-                &QWebEnginePage::loadFinished,
+                &QWebEnginePage::loadStarted,
                 this,
                 &Editor::on_javaScriptWindowObjectCleared);
```

Comment:
This event was called loadFinished and was connected to on_javaScriptWindowObjectCleared.
Now it is called loadStarted and it is still connected to on_javaScriptWindowObjectCleared.

Are we sure the semantic is still correct?